Chummer/Classes/clsCustomDataDirectoryInfo.cs

```diff
@@ -0,0 +1,475 @@
+
+            return intHash;
+        }*/
+        #endregion
+
+        /// <summary>
+        /// Loads and gets all relevant infos from the manifest.xml and stores them in the respective property
+        /// </summary>
+        private void Create()
+        {
+            if (_blnCreated)
+                return;
+
+            try
+            {
+                XmlDocument xmlObjManifest = new XmlDocument();
+                string strFullDirectory = System.IO.Path.Combine(Path, "manifest.xml");
+
+                //Hash = CalculateHash();
+
+                if (!File.Exists(strFullDirectory))
+                    return;
+
+                xmlObjManifest.LoadStandard(strFullDirectory);
+
+                var xmlNode = xmlObjManifest.SelectSingleNode("manifest");
+                xmlNode.TryGetInt32FieldQuickly("version", ref _version);
+
+                xmlNode.TryGetGuidFieldQuickly("guid", ref _guid);
+
+                GetManifestDescriptions(xmlNode);
+                GetManifestAuthors(xmlNode);
+                GetDependencies(xmlNode);
+                GetExclusivities(xmlNode);
+            }
+            catch (Exception ex)
+            {
+                Program.MainForm.ShowMessageBox(
+                    string.Format(GlobalOptions.CultureInfo,
+                        LanguageManager.GetString("Message_FailedLoad"), ex.Message),
+                    string.Format(GlobalOptions.CultureInfo,
+                        LanguageManager.GetString("MessageTitle_FailedLoad")
+                        + " " + Name),
+                    MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+            finally
+            {
+                _blnCreated = true;
+            }
+        }
+        #endregion
+
+        /// <summary>
+        /// Checks if any custom data is activated, that matches name and version of the dependent upon directory
+        /// </summary>
+        /// <param name="objCharacterOptions"></param>
+        /// <returns>List of the names of all missing dependencies as a single string</returns>
+        public string CheckDependency(CharacterOptions objCharacterOptions)
+        {
+            var missingDependencyList = new List<string>();
+            foreach (var (depGuid, depName, depVersion, isMinimumVersion) in DependenciesList)
+            {
+                if (isMinimumVersion)
+                {
+                    if (objCharacterOptions.EnabledCustomDataDirectoryInfos.All(loadedFile =>
+                        depGuid != loadedFile.Guid && depVersion != loadedFile.Version))
+                        missingDependencyList.Add(string.Format(depName + LanguageManager.GetString("AddMinimumVersion"), depVersion));
+                }
+                else
+                {
+                    if (objCharacterOptions.EnabledCustomDataDirectoryInfos.All(loadedFile =>
+                        depGuid != loadedFile.Guid && depVersion >= loadedFile.Version))
```

Comment:
The version conditionals look weird to me here. Right now, `isMinimumVersion == true` will show an error if `depVersion != loadedFile.Version`, while `isMinimumVersion == false` will show one if `depVersion >= loadedFile.Version`. To me, it seems like what should happen is `isMinimumVersion == true` should show an error if `depVersion < loadedFile.Version`, and `isMinimumVersion == false` should show one if `depVersion != loadedFile.Version`.

Reply:
Looking back at them, they are truly a mess right now and will be completely replaced with the new version handling we are discussing down below.